script.js

```diff
@@ -1,100 +1,332 @@
+  const cantidadProduccion = parseInt(document.getElementById("cantidadTotal").value);
+
+  // se valida si hay al menos un material o gasto fijo
+  if (calculadora.materiales.length === 0 && calculadora.gastosFijos.length === 0) {
+    const modalAntesDeCalcular = new bootstrap.Modal(document.getElementById('modalAntesDeCalcular'));
+    modalAntesDeCalcular.show();
+    return;
+  }
+
+  // se validan datos de ganancia y cantidad de producción
+  if (!isNaN(gananciaPorcentual) && !isNaN(cantidadProduccion) && cantidadProduccion > 0) {
+    const resumen = calculadora.generarResumen(gananciaPorcentual, cantidadProduccion);
+
+    const totalVariables = calculadora.calcularCostoMaterialesTotal(cantidadProduccion);
+    const totalFijos = calculadora.calcularGastosFijos();
+    const costoTotal = totalVariables + totalFijos;
+    const costoTotalConGanancia = costoTotal * (1 + gananciaPorcentual / 100);
+    const costoUnidad = costoTotalConGanancia / cantidadProduccion;
+
+    totalVariablesDOM.textContent = `$${totalVariables.toFixed(2)}`;
+    totalFijosDOM.textContent = `$${totalFijos.toFixed(2)}`;
+    costoTotalDOM.textContent = `$${costoTotalConGanancia.toFixed(2)}`;
+    costoGananciaDOM.textContent = `$${costoUnidad.toFixed(2)}`;
+
+    const modalFinal = new bootstrap.Modal(document.getElementById('calculosFinales'));
+    modalFinal.show();
+  }
+});
+
+// Vacía completamente la lista de materiales después de una confirmación del usuario
+document.getElementById("vaciar-materiales").addEventListener("click", function () {
+  if (calculadora.materiales.length === 0) {
+    return; 
   }
-    break;
   
-    case "4":
-      if (calculadora.materiales.length === 0 || calculadora.gastosFijos.length === 0) {
-          alert("Debe agregar al menos un material y un gasto fijo antes de calcular.");
-        } else if (gananciaPorcentual === null || cantidadProduccion === null) {
-          alert("Debe ingresar la ganancia y la cantidad a producir antes de calcular.");
-        } else {
-          const resumen = calculadora.generarResumen(gananciaPorcentual, cantidadProduccion);
-          alert(resumen);
-        }
-        break;
-
-         case "5":
-        salir = true;
-        alert("Gracias por usar CostoPro.");
-        break;
-
-      default:
-        alert("Opción inválida. Ingrese un número del 1 al 5.");
-        break;
-}
+  const filaConfirmacion = document.createElement("tr");
+  filaConfirmacion.innerHTML = `
+    <td colspan="4" class="text-center text-danger">
+      ¿Seguro que quieres vaciar todos los materiales?
+      <button class="btn btn-danger btn-sm btn-confirmar-vaciar">Confirmar</button>
+      <button class="btn btn-secondary btn-sm btn-cancelar-vaciar">Cancelar</button>
+    </td>
+  `;
+  tablaMateriales.innerHTML = "";
+  tablaMateriales.appendChild(filaConfirmacion);
+
+  filaConfirmacion.querySelector(".btn-confirmar-vaciar").addEventListener("click", () => {
+    calculadora.materiales = [];
```

Comment:
Delega la responsabilidad a calculadora:
```suggestion
calculadora.vaciarMateriales() 
```